src/TwigComponent/doc/index.rst

```diff
@@ -468,12 +468,14 @@ component use a ``PreMount`` hook::
         {
             // validate data
             $resolver = new OptionsResolver();
+            $resolver->setIgnoreUndefined(true);
+
             $resolver->setDefaults(['type' => 'success']);
             $resolver->setAllowedValues('type', ['success', 'danger']);
             $resolver->setRequired('message');
             $resolver->setAllowedTypes('message', 'string');
 
-            return $resolver->resolve($data);
+            return $resolver->resolve($data) + $data;
```

Comment:
Should we:
1. add code comment about why we do that?
2. add DOC paragraph about why we do that?
3. stay as is.

Reply:
I think i'd add a entire block of code then, and let this one as it is to not confuse anyone.

Something like 

- this block 
- bloc "info" or "tip" in rst with a short paragraph and just 3 lignes of code 

```php
            $resolver = new OptionsResolver();
            $resolver->setIgnoreUndefined(true);
            
            // ... 
            
            return $resolver->resolve($data) + $data;
```

What do you think ?

Reply:
I've tried to be "short", trust me 😭

Reply:
No i mean it's perfect, but i think having a dedicated block would be more limpid!